gcloud-java-storage/src/main/java/com/google/gcloud/storage/Blob.java

```diff
@@ -456,13 +457,40 @@ public WriteChannel writer(BlobWriteOption... options) {
    * Generates a signed URL for this blob. If you want to allow access to for a fixed amount of time
    * for this blob, you can use this method to generate a URL that is only valid within a certain
    * time period. This is particularly useful if you don't want publicly accessible blobs, but don't
-   * want to require users to explicitly log in.
+   * want to require users to explicitly log in. Signing a URL requires a service account
+   * and its associated key. If a {@link AuthCredentials.ServiceAccountAuthCredentials} was passed
+   * to {@link StorageOptions.Builder#authCredentials(AuthCredentials)} or the default credentials
+   * are being used and the environment variable {@code GOOGLE_APPLICATION_CREDENTIALS} is set, then
+   * {@code signUrl} will use that service account and associated key to sign the URL. If this
```

Comment:
I think we can clarify the precedence of credentials specification a bit more here.  If the user supplies a SignUrlOption with service credentials, we [use those](https://github.com/GoogleCloudPlatform/gcloud-java/blob/master/gcloud-java-storage/src/main/java/com/google/gcloud/storage/StorageImpl.java#L540) instead of the service options' credentials.  If it isn't specified, then we turn to the other options.